### DISCUSSION TITLE
What's the point of cursor text color if the cursor is a beam?

### DISCUSSION THREAD

**Discussion by bound-variable** [kovidgoyal/kitty#5266] [Q&A] [answered]
The [manual](https://sw.kovidgoyal.net/kitty/conf/?highlight=cursor#opt-kitty.cursor_text_color) tells how to configure the color of text under a cursor. But this doesn't have any effect if the cursor is a beam, which is the default if shell integration is enabled. Meanwhile, my text editor (Kakoune) has the colors for the cursor and text configured within itself, not by Kitty.

What's the purpose of defining the text's color if the cursor is a beam?

**Comment by kovidgoyal** [comment]
cursor color most definitely affects the beam. The relevant setting is
cursor, not cursor_text_color. The latter controls the color of the
*text under the cursor* not the color fo the cursor.

On Tue, Jul 12, 2022 at 11:38:43AM -0700, Alan wrote:
> The [manual](https://sw.kovidgoyal.net/kitty/conf/?highlight=cursor#opt-kitty.cursor_text_color) tells how to configure the color of text under a cursor. But this doesn't have any effect if the cursor is a beam, which is the default if shell integration is enabled. Meanwhile, my text editor (Kakoune) has the colors for the cursor and text configured within itself, not by Kitty.
> 
> What's the purpose of defining the text's color if the cursor is a beam?
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/5266
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by page-down** [comment, marked as answer]
> ... doesn't have any effect if the cursor is a beam, which is the default ...
> What's the purpose of defining the text's color ...

~I see it as the fallback color.~
This option is only applied to the block cursor.

> ... Meanwhile, my text editor ... has the colors ... configured ...

Your text editor can set the cursor to block, if desired, at which point the cursor text color will be reflected.

**Reply by bound-variable** [reply]
Kakoune reads color specs for text from its config. Can it then also be considered as fallback in such cases?

Kakoune does have a default value for the text and cursor, but my understanding is that this reads _foreground_ and _background_, not _cursor text color_.

**Reply by page-down** [reply]
My previous statement is somewhat incorrect. If I remember correctly, cursor text color cannot be set independently by escape sequence.

In the terminal, there are "foreground color" and "background color" for text.
A program can freely change the fg color and bg color of the text, can read the cursor position and change the text color at the cursor position.
In kitty, you can specify the cursor and cursor text colors, and you can also configure to reverse the foreground and background colors at the cursor position (reverses the color specified by the program).

> What's the point of cursor text color if the cursor is a beam?

The purpose of this option is to set the text foreground color for any program that uses the block cursor.
The shell integration enabled by default sets the cursor to _beam_, which does not mean that the block cursor will never appear in the terminal.

You can see in the default configuration that kitty's default fallback cursor shape is **block**, not _beam_.

**Reply by bound-variable** [reply]
> The purpose of this option is to set the text foreground color for any program that uses the block cursor.

Sorry, what I'm trying to say is that Kakoune can set the colors of the block cursor and the text when using block cursor. So, in that case, kitty's setting is used at the command prompt, or else is superfluousl, right? It's not used in the text editor.

**Reply by page-down** [reply]
I have never used kakoune. As far as I know, kakoune hides the cursor in most cases, records the edit position internally and simulates a block cursor by changing the fg and bg of the text.

In this case, it has nothing to do with the `cursor` or `cursor_text_color` configuration option.

**Reply by bound-variable** [reply]
Okay, great. Thank you!

**Comment by kovidgoyal** [comment]
cursor is the option that sets the cursor color and it most definitely applies to beam.
